src/main/kotlin/com/example/passmanager/migration/InitialIndexCreationChangeUnit.kt

```diff
@@ -26,36 +25,37 @@ internal class InitialIndexCreationChangeUnit(private val mongoTemplate: MongoTe
                 .append(MongoPass::passOwnerId.name, 1)
                 .append(MongoPass::purchasedAt.name, 1)
         )
-        mongoTemplate.indexOps<MongoPass>().ensureIndex(passOwnerIdAndPurchasedAtIndexDefinition).also { indexName ->
-            passOwnerIdAndPurchasedAtIndexName = indexName
-        }
-        passOwnerIndexNames = mongoTemplate.indexOps<MongoPassOwner>().run {
-            listOf(
-                ensureIndex(Index(MongoPassOwner::email.name, ASC).unique()),
-                ensureIndex(Index(MongoPassOwner::phoneNumber.name, ASC).unique())
-            )
+        mongoTemplate.indexOps<MongoPass>().ensureIndex(passOwnerIdAndPurchasedAtIndexDefinition).subscribe()
+
+        mongoTemplate.indexOps<MongoPassOwner>().apply {
+            ensureIndex(Index(MongoPassOwner::email.name, ASC).unique()).subscribe()
+            ensureIndex(Index(MongoPassOwner::phoneNumber.name, ASC).unique()).subscribe()
         }
     }
 
     @RollbackExecution
     fun dropIndexesInCaseOfFailure() {
         log.info("Rolling back ${this::class.simpleName}")
 
-        val allPassIndexNames = indexNames<MongoPass>()
-        if (allPassIndexNames.contains(passOwnerIdAndPurchasedAtIndexName)) {
-            mongoTemplate.indexOps<MongoPass>().dropIndex(passOwnerIdAndPurchasedAtIndexName)
-        }
-
-        val passOwnerIndexOperations = mongoTemplate.indexOps<MongoPassOwner>()
-        passOwnerIndexNames.filter { it in indexNames<MongoPassOwner>() }
-            .forEach { passOwnerIndexOperations.dropIndex(it) }
+        Flux.merge(indexNamesFlux<MongoPass>(), indexNamesFlux<MongoPassOwner>())
+            .filter { indexNameToCollectionNameMap.containsKey(it) }
+            .flatMap { indexName ->
+                val collectionName = indexNameToCollectionNameMap[indexName].orEmpty()
+                mongoTemplate.indexOps(collectionName).dropIndex(indexName)
```

Comment:
We create 2 indexes but remove 3.

Is it ok?

Reply:
We create 3 :)

```
mongoTemplate.indexOps<MongoPass>().ensureIndex(passOwnerIdAndPurchasedAtIndexDefinition).subscribe()

mongoTemplate.indexOps<MongoPassOwner>().apply {
     ensureIndex(Index(MongoPassOwner::email.name, ASC).unique()).subscribe()
      ensureIndex(Index(MongoPassOwner::phoneNumber.name, ASC).unique()).subscribe()
}
```